src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/CookieOptionsConfiguration.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Authentication.AzureAD.UI;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.Options;
+using OrchardCore.Modules;
+
+namespace OrchardCore.Microsoft.Authentication.Configuration
+{
+    internal class CookieOptionsConfiguration : IConfigureNamedOptions<CookieAuthenticationOptions>
+    {
+        private readonly IOptionsMonitor<AzureADOptions> _azureADOptions;
+
+        public CookieOptionsConfiguration(IOptionsMonitor<AzureADOptions> azureADOptions)
+        {
+            _azureADOptions = azureADOptions;
+        }
+
+        public void Configure(string name, CookieAuthenticationOptions options)
+        {
+            if (name != "Identity.External")
+            {
+                return;
+            }
+
+            options.LoginPath = $"/AzureAD/Account/SignIn/{AzureADDefaults.AuthenticationScheme}";
```

Comment:
Can it handle `~/` instead to support multi-tenancy?

Reply:
i used the options.Cookie.Path in order to isolate Identity.External cookie per tenant and prevent correlation errors

Reply:
I mean that I see a url here. How is LoginPath used such that the final link or redirection will actually stay on the same tenant?

Reply:
I modified the url path to include ~ and added the Cookie.Path to isolate the cookies also :)